a2rchi/cli/cli_main.py

```diff
@@ -360,19 +363,22 @@ def create(
     # prepare grader service if requested
     compose_template_vars["use_grader_service"] = use_grader_service
     if use_grader_service:
-        _print_msg("Preparing Grader Service")
+        if not dry:
+            _print_msg("Preparing Grader Service")
         compose_template_vars["grader_image"] = f"grader-{name}"
         compose_template_vars["grader_tag"] = tag
 
         compose_template_vars["grader_volume_name"] = f"a2rchi-grader-{name}"
-        _create_volume(compose_template_vars["grader_volume_name"], podman=use_podman)
+        if not dry:
+          _create_volume(compose_template_vars["grader_volume_name"], podman=use_podman)
 
         _prepare_secret(a2rchi_name_dir, "admin_password", locations_of_secrets)
```

Comment:
do we want to prepare secrets in --dry mode? Similar as above. It's sort of up to what we decide we mean by "dry" mode. What is most useful? What is the use case for this? If it's just to dump the config, I would think we don't need to preare secrets, make dirs, etc. But maybe you intended to want to check that all those things are being done correctly, without creating volumes. In any case, let's decide, and describe it more clearly in the parser.